src/hooks/use-field-id/use-field-id.js

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import getFieldId from '../../utils/get-field-id';
+
+const useFieldId = (id, sequentialId) => {
+  const [internalId, setId] = React.useState(id);
+
+  React.useEffect(() => {
+    setId(getFieldId({ id }, { id: internalId }, sequentialId));
```

Comment:
Sorry, I fail to follow. Why all the arguments? What does this do?

Reply:
```js
const getFieldId = (props, state, createId) =>
  props.id || state.id || createId();

export default getFieldId;
```

so if props id set set, it uses that, else it uses the internal state one, if that's not set, it uses sequentialId to create a new id